### PR TITLE
(CDAP-14986) Correctly setup singleton bindings for storage module

### PR DESCRIPTION
**Pull request by chtyim**

### CODE REVIEW

**Comment by chtyim** [comment]
Build https://builds.cask.co/browse/CDAP-DUT6888

File: cdap-data-fabric/src/main/java/co/cask/cdap/data/runtime/StorageModule.java [hunk 1]
```diff
@@ -45,10 +44,11 @@
 
   @Override
   protected void configure() {
-    bind(TransactionRunner.class).toProvider(TransactionRunnerProvider.class);
-    bind(StructuredTableAdmin.class).toProvider(StructuredTableAdminProvider.class);
-    bind(StructuredTableRegistry.class).toProvider(StructuredTableRegistryProvider.class);
-    bind(DataSourceInstantiator.class).in(Singleton.class);
+    bind(TransactionRunner.class).toProvider(TransactionRunnerProvider.class).in(Scopes.SINGLETON);
```
Comment (yaojiefeng) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data/runtime/StorageModule.java:47]: Do we need the `TransactionRunner` and `StructuredTableAdmin` also in singleton?

Comment (chtyim) [reply cdap-data-fabric/src/main/java/co/cask/cdap/data/runtime/StorageModule.java:47]: I do this based on the original Singleton annotation on the provider classes, which I assume the intention was to have all these bindings Singleton ?

Comment (yaojiefeng) [reply cdap-data-fabric/src/main/java/co/cask/cdap/data/runtime/StorageModule.java:47]: I think I misunderstand this line of code. To clarify, will this just make the provider singleton? If I call `get()` method, will the class returned will not be in singleton, right?

Comment (anew) [reply cdap-data-fabric/src/main/java/co/cask/cdap/data/runtime/StorageModule.java:47]: haha that is a nice ambiguity in the Guice API :) I always assumed that the instances becomes a singleton, not the provider. But maybe I have been wrong all this time.

Comment (anew) [reply cdap-data-fabric/src/main/java/co/cask/cdap/data/runtime/StorageModule.java:47]: I think it is answered here: https://stackoverflow.com/questions/25472785/how-to-get-guice-to-provide-a-unique-instance-with-a-provider

**Review by anew — APPROVED** [review]

File: cdap-data-fabric/src/main/java/co/cask/cdap/data/runtime/StorageModule.java (hunk 1 above)
Comment (chtyim) [reply cdap-data-fabric/src/main/java/co/cask/cdap/data/runtime/StorageModule.java:47]: The `in(Scope)` always target the source of the binding, so it doesn't matter what it binds to.

**Pull request merged**